RocketChat/Rocket.Chat
# Regression: Fix Default Business hour overriding other Business Hours

## Proposed changes (including videos or screenshots)





## Issue(s)

There was a bug introduced in this PR - https://github.com/RocketChat/Rocket.Chat/pull/24112 

## Steps to test or reproduce



Assumption: Business Hour is activated & is set to multiple

1. An agent "xyz" is associated with Department A.
2. Make Business Hours for Department A are closed.
3. Make sure "Default Business Hour" is open
4. Login as an agent "xyz" from Department A.
5. Click on the omnichannel icon to make the agent available.

Expected Behaviour: Agent shouldn't be allowed to make themself available

Actual Behaviour: Agents are able to make themself available

## Further comments
